lib/class.OS_Linux.php

```diff
@@ -935,18 +935,23 @@ public function getBattery() {
 		// Get vals for each battery
 		foreach ($bats as $b) {
 
-			$go_for_it = true;
-
-			foreach(array($b.'/manufacturer', $b.'/status', $b.'/charge_now') as $f)
+			foreach(array($b.'/manufacturer', $b.'/status') as $f)
 				if (!is_file($f))
-					$go_for_it = false; // Continue out of two nested loops
-
-			if (!$go_for_it) 
-				continue;
+					continue 2;
 
 			// Get these from the simple text files
-			$charge_full = LinfoCommon::getIntFromFile($b.'/charge_full');
-			$charge_now = LinfoCommon::getIntFromFile($b.'/charge_now');
+			switch (true) {
+				case is_file($b.'/energy_full'):
+					$charge_full = LinfoCommon::getIntFromFile($b.'/energy_full');
+					$charge_now = LinfoCommon::getIntFromFile($b.'/energy_now');
+					break;
+				case is_file($b.'/charge_full'):
+					$charge_full = LinfoCommon::getIntFromFile($b.'/charge_full');
+					$charge_now = LinfoCommon::getIntFromFile($b.'/charge_now');
+					break;
+				default:
+					continue;
```

Comment:
One last thing, promise! Can you please put a break after this continue?

It isn't strictly needed but it does aid readability and will reduce the chances of bad things happening later if more code is added to the switch's body.